File: Programmers/pro_60057_문자열 압축.py
def solution(s):
    answer = len(s)    
    for l in range(1,len(s)//2+2):
        temp = ''
        idx, cnt = 0, 0
        while(idx<len(s)):
            cur = s[idx:idx+l]
            if cur == s[idx+l:idx+2*l]:
                cnt+=1
            else:
                if(cnt>0):
                    temp+=str(cnt+1)
                    cnt=0
                temp+=cur
            idx+=l
        answer=min(answer,len(temp))
    return answer

File: Programmers/pro_60057_문자열압축.js
function solution(s) {
    const len = s.length
    var answer = len;
    for (let i = 1; i < len / 2 + 1; i++) {
        let str = ''
        let [idx, cnt] = [0, 0]
        while (idx <= len) {
            let cur = s.slice(idx, idx + i)
            if (cur == s.slice(idx + i, idx + 2 * i)) {
                cnt += 1
            } else {
                if (cnt > 0) {
                    str += (cnt + 1) + ""
                    cnt = 0
                }
                str += cur
            }
            idx += i
        }
        answer = Math.min(str.length, answer)
    }
    return answer;
}